kolibri/plugins/learn/assets/src/views/ChannelRenderer/CustomContentRenderer.vue

```diff
@@ -114,19 +132,33 @@
       // called in mainClient.js
       fetchContentCollection(message) {
         const { options } = message;
+        const { kinds, onlyContent, onlyTopics } = options;
+
+        if (onlyContent && onlyTopics) {
+          const err = new Error('onlyContent and onlyTopics can not be used at the same time');
+          return createReturnMsg({ message, err });
+        }
+
         return ContentNodeResource.fetchCollection({
           getParams: {
             ids: options.ids,
+            authors: options.authors,
+            tags: options.tags,
             parent: options.parent === 'self' ? this.topic.id : options.parent,
+            channel_id: this.topic.channel_id,
+            max_results: options.maxResults ? options.maxResults : 50,
+            cursor: options.cursor,
```

Comment:
I think rather than passing `cursor` here - it would be better to define a separate `fetchMore` pathway, and the `more` object that was returned can just be returned back. Otherwise, mixing the cursor with potential other changes to parameters could lead to unexpected results.

Reply:
Okay, I didn't thought about that problem. I've created a new method in the API, `getContentPage`, to request new pages with the cursor and just removed the `cursor` from the `getContentByFilter`.